src/typeahead/typeahead.ts

```diff
@@ -215,7 +216,11 @@ export class NgbTypeahead implements ControlValueAccessor,
   }
 
   onDocumentClick(event) {
-    if (event.target !== this._elementRef.nativeElement) {
+    if (event.target !== this._elementRef.nativeElement &&
+      (!this._windowRef ||
+       !this._windowRef.location ||
+       !this._windowRef.location.nativeElement ||
+       !this._elementClassNameContains(event.target, this._windowRef.location.nativeElement.className))) {
```

Comment:
I'm sorry, but it is not clear for me why we've got all this logic? Are you trying to check if the event is not coming from the results popup? If so, wouldn't check we use in other places (ex. https://github.com/ng-bootstrap/ng-bootstrap/blob/958b955fb3fdc96ff8292c472363adb326035bcf/src/dropdown/dropdown.ts#L31) work?

Reply:
@pkozlowski-opensource sorry, didn't see this until now. :/
Yes this is to check if the event is not coming from the results popup.
Unfortunately your suggested solution will not work on iOS if the results popup contains any other angular component.

For example:
I'm using a typeahead popup to show product results in a search box.
Each row in the typeahead popup is an angular component that shows a product image, product title, article number and price.

When tapping on any of the rows on iOS, it will simply close the popup nomatter what.
The code I provided is so far the only way I have found to make it behave like it should on iOS.